needle.pm

```diff
@@ -81,7 +82,13 @@ sub new {
     $self->{file} = $jsonfile;
     $self->{name} = basename($jsonfile, '.json');
     my $png = $self->{png} || $self->{name} . ".png";
-    $self->{png} = File::Spec->catpath('', dirname($jsonfile), $png);
+
+    my $original_needles_dir = dirname($jsonfile);
+    $self->{png} = File::Spec->catpath('', $original_needles_dir, $png);
+
+    # we want to load needles from the cache if it is present
+    $self->{png} =~ s#$original_needles_dir#$shared_cache#g if ($shared_cache);
```

Comment:
@aaannz Mine too, i just panic when there are slashes in the string i want to replace... anywho, changed it, looks beautiful now :+1: